src/DotNetty.Codecs.Protobuf/ProtobufEncoder.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DotNetty.Codecs.Protobuf
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.Contracts;
+    using DotNetty.Buffers;
+    using DotNetty.Transport.Channels;
+    using Google.Protobuf;
+
+    public sealed class ProtobufEncoder : MessageToMessageEncoder<IMessage>
+    {
+        protected override void Encode(IChannelHandlerContext context, IMessage message, List<object> output)
+        {
+            Contract.Requires(context != null);
+            Contract.Requires(message != null);
+            Contract.Requires(output != null);
+
+            IByteBuffer buffer = null;
+
+            try
+            {
+                int size = message.CalculateSize();
+                if (size == 0)
+                {
+                    return;
+                }
+
+                buffer = Unpooled.WrappedBuffer(message.ToByteArray());
```

Comment:
I guess it depends on message size: better for small, worse for bigger. Might we worth asking them to make ctor with array segment parameters public.